evm/src/arithmetic/operations.rs

```diff
@@ -0,0 +1,156 @@
+use ethereum_types::U256;
+use plonky2::hash::hash_types::RichField;
+
+use crate::arithmetic::columns::*;
+use crate::arithmetic::{add, compare, modular, mul, sub};
+
+#[inline]
+fn u64_to_array<F: RichField>(out: &mut [F], x: u64) {
+    debug_assert!(out.len() == 4);
+
+    const MASK: u64 = (1 << 16) - 1;
+    out[0] = F::from_canonical_u64(x & MASK);
+    out[1] = F::from_canonical_u64((x >> 16) % MASK);
+    out[2] = F::from_canonical_u64((x >> 32) % MASK);
+    out[3] = F::from_canonical_u64((x >> 48) % MASK);
+}
+
+fn u256_to_array<F: RichField>(out: &mut [F], x: U256) {
+    debug_assert!(out.len() == N_LIMBS);
+
+    u64_to_array(&mut out[0..4], x.0[0]);
+    u64_to_array(&mut out[4..8], x.0[1]);
+    u64_to_array(&mut out[8..12], x.0[2]);
+    u64_to_array(&mut out[12..16], x.0[3]);
+}
```

Comment:
Seems like this part of the code depends on `LIMB_BITS=16`. We probably won't change this constant anytime soon, but maybe it's worth adding a couple of `(debug_)assert!`s.